socket.go

```diff
@@ -18,14 +18,16 @@ import (
 )
 
 const (
-	defaultRetry   = 250 * time.Millisecond
-	defaultTimeout = 5 * time.Minute
+	defaultRetry = 250 * time.Millisecond
+	// defaultTimeout = 5 * time.Minute
+	defaultTimeout = 15 * time.Second
 )
 
 var (
 	errInvalidAddress = errors.New("zmq4: invalid address")
 
-	ErrBadProperty = errors.New("zmq4: bad property")
+	ErrBadProperty      = errors.New("zmq4: bad property")
+	ErrUnknownTransport = errors.New("zmq4: unknown transport")
```

Comment:
instead of a value, I'd rather have it as a type, `TransportError` or something, so we could provide the name of the unknown transport.

Reply:
Make it a feature request.
Not related to this PR at all.